src/player/player.rs

```diff
@@ -240,15 +242,50 @@ async fn create_session(
     credentials: Credentials,
     ap_port: Option<u16>,
 ) -> Result<Session, SpotifyError> {
-    let session_config = SessionConfig {
-        ap_port,
-        ..Default::default()
-    };
-    let result = Session::connect(session_config, credentials, None).await;
-    result.map_err(|e| {
-        dbg!(e);
-        SpotifyError::LoginFailed
-    })
+    match ap_port {
+        Some(ap_port) => {
+            let session_config = SessionConfig {
+                ap_port: Some(ap_port),
+                ..Default::default()
+            };
+            let result = Session::connect(session_config, credentials, None).await;
+            result.map_err(|e| {
+                dbg!(e);
+                SpotifyError::LoginFailed
+            })
+        },
+        None => {
+            let mut ret = None;
+
+            let mut ports: Vec<Option<u16>> = KNOWN_AP_PORTS.iter().map(|port| Some(port.to_owned())).collect();
+            ports.push(None);
```

Comment:
Shouldn't the default port be tried first?

Reply:
Fixed in the latest commit (`KNOWN_AP_PORTS`'s first element is now `None`). Thank you for helpful reviews :)